### PR TITLE
Update documentation for TrailingComma rules

### PR DESCRIPTION
**Pull request by chao2zhang**
- This is the first part of addressing https://github.com/detekt/detekt/pull/5386#issuecomment-1304916205
- The second part would be adding `@ActiveByDefault`, however I was running into runtime exception at multiple places across our codebase, hence I would leave it for the next PR.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -166,3 +166,6 @@ target/
 /website/docs/rules/*.md
 /website/static/kdoc
 /website/docs/gettingstarted/_cli-options.md
+
+# Jenv
+.java-version
```
Comment (cortinico) [inline .gitignore:171]: I'm unsure this should be .gitignored. I'm actually in favor of having a `.java-version` in the root folder

**Review by 3flex — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (chao2zhang) [reply .gitignore:171]: Yes but everyone's `.java-version` content will be different depending on the exact minor/patch version

File: detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/TrailingCommaOnCallSite.kt [hunk 2]
```diff
@@ -3,13 +3,17 @@ package io.gitlab.arturbosch.detekt.formatting.wrappers
...
 import io.gitlab.arturbosch.detekt.api.internal.AutoCorrectable
 import io.gitlab.arturbosch.detekt.api.internal.Configuration
 import io.gitlab.arturbosch.detekt.formatting.FormattingRule
 
 /**
  * See [ktlint docs](https://pinterest.github.io/ktlint/rules/standard/) for documentation.
+ *
+ * [Kotlin coding convention](https://kotlinlang.org/docs/coding-conventions.html#trailing-commas) recommends
+ * trailing comma encourage the use of trailing commas at the declaration site and
+ * leaves it at your discretion for the call site.
```
[the remark this answers is not in the document]
Comment (chao2zhang) [reply detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/wrappers/TrailingCommaOnCallSite.kt:16]: Will emphasize ktlint here

File: .gitignore (hunk 1 above)
Comment (3flex) [reply .gitignore:171]: Maybe open a separate PR for this? I'm otherwise happy to merge this PR.

Comment (BraisGabin) [reply .gitignore:171]: I agree that we should move this to another pr. This PR is the last thing that we need to release 1.22.0.

Comment (cortinico) [reply .gitignore:171]: > Yes but everyone's .java-version content will be different depending on the exact minor/patch version

You should be able to have just the major version (like `11`) inside `.java-version` and that will be picked up correctly by jenv

**Comment by cortinico** [comment]
@chao2zhang can you address the comment of @BraisGabin and me about the `.java-version`? Aside from this, we can merge this one 👍

**Review by cortinico — APPROVED** [review]

**Review by schalkms — APPROVED** [review]

**Pull request merged**